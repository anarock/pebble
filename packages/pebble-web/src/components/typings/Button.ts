import * as React from "react";
import * as CSS from "csstype";
import { Interpolation } from "./emotionCustom";

export type ButtonType = "primary" | "secondary" | "link" | "success" | "alert";

export interface ButtonProps {
  children: React.ReactNode;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  showShadow?: boolean;
  disabled?: boolean;
  type?: ButtonType;
  size?: "x-small" | "small" | "large";
  width?: CSS.WidthProperty<number>;
  styles?: Interpolation;
  showRipple?: boolean;
  loading?: boolean;
  filled?: boolean;
  buttonProps?: React.ButtonHTMLAttributes<HTMLButtonElement>;
}

export interface DropDownButtonProps extends ButtonProps {
  isOpen: boolean;
  isSelected: boolean;
}

interface MappingColorByTypeOptions {
  textColor?: string;
  base: string;
  hover: string;
  active: string;
  disabled: string;
}

export interface MappingColorByType {
  primary: MappingColorByTypeOptions;
  secondary: MappingColorByTypeOptions;
  success: MappingColorByTypeOptions;
  alert: MappingColorByTypeOptions;
}
