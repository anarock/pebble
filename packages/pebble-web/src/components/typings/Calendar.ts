import { CalendarProps as CP } from "react-calendar";
import { Omit } from "utility-types";
import { Interpolation } from "./emotionCustom";

interface TileDot {
  timeStamp?: number | Date;
  colors?: string[];
}

interface CommonCalendarProps extends Omit<CP, "onChange"> {
  hideShadow?: boolean;
  wrapperStyles?: Interpolation;
  wrapperClassName?: string;
  onClear?: () => void;
  tileDots: TileDot[];
  disabledDays?: Array<number | Date>;
}

export interface DateSingle extends CommonCalendarProps {
  range?: false;
  selected?: Date;
  onChange: (value: Date) => void;
  onApply?: (value?: Date) => void;
}

export interface DateRange extends CommonCalendarProps {
  range: true;
  selected?: [Date, Date];
  onChange: (value: [Date, Date]) => void;
  onApply?: (value?: [Date, Date]) => void;
}

export type CalendarProps = DateSingle | DateRange;

export interface CalendarState {
  value?: [Date, Date] | Date;
  singleSelectedDate?: [Date, Date] | null;
}
