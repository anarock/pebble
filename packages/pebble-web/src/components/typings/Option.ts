import { ControlProps } from "./Control";
import { Omit } from "utility-types";
import * as React from "react";
import { Interpolation } from "./emotionCustom";

export interface OptionProps<OptionType>
  extends Omit<ControlProps<OptionType>, "checked" | "type"> {
  isActive?: boolean;
  isSelected?: boolean;
  multiSelect?: boolean;
  labelStyles?: Interpolation;
  rightElement?: (props: OptionProps<OptionType>) => React.ReactNode;
}
