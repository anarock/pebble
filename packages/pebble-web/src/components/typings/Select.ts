import { SimpleInputProps } from "./Input";
import { SearchProps } from "./Search";
import { Extras } from "./OptionGroup";
import { Omit } from "utility-types";
import { Placement, Modifiers } from "popper.js";
import { Interpolation } from "./emotionCustom";

export type SingleSelected = number | string;
export type MultiSelected = Array<number | string>;

interface CommonSelectProps<OptionType> {
  styles?: Interpolation;
  placeholder: string;
  required?: boolean;
  errorMessage?: string;
  value?: string;
  onClear?: () => void;
  searchBox?: boolean;
  searchBoxProps?: Omit<SearchProps, "type">;
  dropdownStyles?: Interpolation;
  inputProps?: Omit<SimpleInputProps, "onChange" | "value" | "placeholder">;
  fullWidthDropdown?: boolean;
  onDropdownToggle?: (isOpen: boolean) => void;
  disabled?: boolean;
  children: React.ReactNode;
  isSelected?: (value: OptionType) => boolean;
  placement?: Placement;
  modifiers?: Modifiers;
}

export interface SingleSelectProps<OptionType>
  extends CommonSelectProps<OptionType> {
  multiSelect?: false;
  onChange: (value: OptionType, extras: Extras) => void;
  /**
   * @deprecated use isSelected
   */
  selected?: OptionType;
}

export interface MultiSelectProps<OptionType>
  extends CommonSelectProps<OptionType> {
  multiSelect: true;
  onChange: (value: OptionType[], extras: Extras) => void;
  onApply?: (value: OptionType[], props: SelectProps<OptionType>) => void;
  /**
   * @deprecated use isSelected
   */
  selected?: OptionType[];
}

export type SelectProps<OptionType> =
  | SingleSelectProps<OptionType>
  | MultiSelectProps<OptionType>;
