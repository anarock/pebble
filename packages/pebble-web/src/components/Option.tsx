import * as React from "react";
import { OptionProps } from "./typings/Option";
import Control from "./shared/Control";
import {
  activeRow,
  rowWrapper,
  selectedRow,
  labelWrap
} from "./styles/Options.styles";
import Ink from "react-ink";
import { colors } from "pebble-shared";

const defaultProps = {
  rightElement: ({
    isSelected,
    multiSelect
  }: {
    isSelected: boolean;
    multiSelect: boolean;
  }) => {
    const iconClass = [
      "pi",
      isSelected && "pi-checkbox-selected",
      !isSelected && "pi-checkbox-unselected"
    ]
      .filter(Boolean)
      .join(" ");
    return multiSelect ? (
      <i
        className={iconClass}
        css={{
          marginLeft: "10px",
          color: isSelected ? colors.violet.base : colors.gray.light,
          fontSize: "20px"
        }}
      />
    ) : null;
  }
};

class Option<OptionType> extends React.Component<
  OptionProps<OptionType> & Required<typeof defaultProps>
> {
  static defaultProps = defaultProps;
  render() {
    const {
      label,
      isActive,
      isSelected,
      rightElement,
      labelStyles,
      styles
    } = this.props;
    const _class = [
      rowWrapper,
      !!isActive && activeRow,
      !!isSelected && selectedRow,
      styles
    ];
    return (
      <Control
        {...this.props}
        checked={this.props.isSelected}
        type={this.props.multiSelect ? "checkbox" : "radio"}
        styles={_class}
      >
        {() => {
          return (
            <>
              <div css={[labelWrap, labelStyles]}>{label}</div>
              {rightElement(this.props)}
              <Ink />
            </>
          );
        }}
      </Control>
    );
  }
}

export default Option;
