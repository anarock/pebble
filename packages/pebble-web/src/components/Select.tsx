import * as React from "react";

import { SelectProps } from "./typings/Select";
import {
  chevronStyle,
  dropDownClass,
  inputWrapper,
  selectInput,
  selectInputWrapper,
  selectWrapper,
  fullWidth,
  relativePosition
} from "./styles/Select.styles";
import DropDown from "./DropDown";
import Input from "./Input";
import OptionGroupCheckBox from "./OptionGroupCheckBox";
import OptionGroupRadio from "./OptionGroupRadio";

function noop() {}

function Select<OptionType>(props: SelectProps<OptionType>) {
  const {
    styles,
    placeholder,
    required,
    errorMessage,
    value,
    dropdownStyles,
    inputProps,
    fullWidthDropdown,
    onDropdownToggle = noop,
    disabled,
    isSelected,
    placement,
    modifiers
  } = props;

  return (
    <div css={[selectWrapper, styles, !!fullWidthDropdown && relativePosition]}>
      <DropDown
        dropDownStyles={[
          dropDownClass,
          dropdownStyles,
          !!fullWidthDropdown && fullWidth
        ]}
        onOutsideClick={isOpen => onDropdownToggle(isOpen)}
        labelComponent={({ toggleDropdown, isOpen }) => {
          const chevron = [
            "pi",
            "pi-arrow-drop-down",
            isOpen && "__pebble__select__open"
          ]
            .filter(Boolean)
            .join(" ");
          return (
            <div
              css={inputWrapper}
              onClick={
                disabled
                  ? undefined
                  : () => {
                      toggleDropdown();
                      onDropdownToggle(isOpen);
                    }
              }
            >
              <Input
                styles={selectInputWrapper}
                inputStyles={selectInput}
                placeholder={placeholder}
                value={value || ""}
                onChange={noop}
                required={required}
                message={isOpen ? " " : ""}
                errorMessage={errorMessage}
                readOnly
                disabled={disabled}
                {...inputProps}
              />
              <i css={chevronStyle} className={chevron} />
            </div>
          );
        }}
        placement={placement}
        modifiers={modifiers}
      >
        {({ toggle, isOpen }) => {
          const { children, onClear, searchBox, searchBoxProps } = props;
          const commonProps = {
            isSelected,
            onClear:
              onClear &&
              (() => {
                onClear();
                onDropdownToggle(isOpen);
                toggle();
              }),
            searchBox,
            searchBoxProps
          };

          // This would have been the ideal way to write this but typescript is crying.
          // const OptionGroup = props.multiSelect
          //   ? OptionGroupCheckBox
          //   : OptionGroupRadio;
          // return (
          //   <OptionGroup
          //     selected={props.selected}
          //     onChange={(_value, extras) => {
          //       if (_value) {
          //         props.onChange(_value, extras);
          //       }
          //       if (!props.multiSelect) {
          //         onDropdownToggle(isOpen);
          //         toggle();
          //       }
          //     }}
          //     onApply={
          //       props.multiSelect &&
          //       props.onApply &&
          //       (_value => {
          //         if (props.onApply) {
          //           props.onApply(_value, props);
          //         }
          //         onDropdownToggle(isOpen);
          //         toggle();
          //       })
          //     }
          //     {...commonProps}
          //   >
          //     {children}
          //   </OptionGroup>
          // );

          if (props.multiSelect) {
            return (
              <OptionGroupCheckBox<OptionType>
                selected={props.selected}
                onChange={(_value, extras) => {
                  props.onChange(_value, extras);
                }}
                onApply={
                  props.onApply &&
                  (_value => {
                    if (props.onApply) props.onApply(_value, props);
                    onDropdownToggle(isOpen);
                    toggle();
                  })
                }
                {...commonProps}
              >
                {children}
              </OptionGroupCheckBox>
            );
          } else {
            return (
              <OptionGroupRadio
                selected={props.selected}
                onChange={(_value, extras) => {
                  if (_value) props.onChange(_value, extras);
                  onDropdownToggle(isOpen);
                  toggle();
                }}
                {...commonProps}
              >
                {children}
              </OptionGroupRadio>
            );
          }
        }}
      </DropDown>
    </div>
  );
}

export default Select;
