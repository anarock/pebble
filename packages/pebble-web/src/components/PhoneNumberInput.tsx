import * as React from "react";
import Select from "./Select";
import Input from "./Input";

import { PhoneNumberInputProps } from "./typings/PhoneNumberInput";
import {
  wrapper,
  selectStyle,
  combinedLabelStyle
} from "./styles/PhoneNumberInput.styles";
import { labelStyle } from "./styles/Input.styles";
import { colors } from "pebble-shared";

export default class PhoneNumberInput extends React.Component<
  PhoneNumberInputProps
> {
  onCountrySelect = (countryCode: string) => {
    this.props.onChange({
      countryCode,
      phone: this.props.phone
    });
  };

  onNumberChange = (value: string) => {
    const _value = value.replace(/\D/g, "");
    if (_value === this.props.phone) {
      return;
    }
    this.props.onChange({
      countryCode: this.props.countryCode,
      phone: _value
    });
  };

  render() {
    const {
      phone,
      countryCode,
      styles,
      selectProps,
      inputProps,
      required,
      placeholder
    } = this.props;
    return (
      <div css={[wrapper, styles]}>
        <label
          className="_pebble_input_label_focused"
          css={[labelStyle, combinedLabelStyle]}
        >
          {placeholder || "Phone No."}
          {required && <span style={{ color: colors.red.base }}>&nbsp;*</span>}
        </label>
        <Select
          placeholder=""
          onChange={this.onCountrySelect}
          value={countryCode + ""}
          selected={countryCode}
          {...selectProps}
          styles={[selectStyle, selectProps && selectProps.styles]}
        >
          {this.props.children}
        </Select>
        <Input
          onChange={this.onNumberChange}
          placeholder=""
          value={phone}
          {...inputProps}
        />
      </div>
    );
  }
}
