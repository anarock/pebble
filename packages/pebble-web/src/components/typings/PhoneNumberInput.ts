import { SingleSelectProps } from "./Select";
import { SimpleInputProps } from "./Input";
import { Omit } from "utility-types";
import { Interpolation } from "./emotionCustom";

export interface PhoneNumberInputProps {
  countryCode: string;
  phone: string;
  onChange: ({
    countryCode,
    phone
  }: {
    countryCode: string;
    phone: string;
  }) => void;

  // Optional
  styles?: Interpolation;
  selectProps?: Partial<SingleSelectProps<string>>;
  inputProps?: Omit<SimpleInputProps, "value" | "onChange" | "placeholder">;
  placeholder?: string;
  required?: boolean;
}
