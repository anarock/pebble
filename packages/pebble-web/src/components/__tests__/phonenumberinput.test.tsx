import * as React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";

import { PhoneNumberInputProps } from "../typings/PhoneNumberInput";
import Option from "../Option";
import Select from "../Select";
import PhoneNumberInput from "../PhoneNumberInput";
import countries from "./fixtures/countrycodes";

const SELECT_INPUT_ID = "phone-select-input-test";
const PHONE_INPUT_ID = "phone-test-input-test";

function getComponent(props: Partial<PhoneNumberInputProps> = {}) {
  return (
    <PhoneNumberInput
      onChange={() => {}}
      phone=""
      countryCode="+1"
      {...props}
      selectProps={{
        inputProps: {
          textArea: false,
          inputProps: {
            id: SELECT_INPUT_ID
          }
        }
      }}
      inputProps={{
        inputProps: {
          id: PHONE_INPUT_ID
        }
      }}
    >
      {countries.map(country => (
        <Option
          key={country.id}
          value={country.country_code}
          label={`${country.name} (${country.country_code})`}
        />
      ))}
    </PhoneNumberInput>
  );
}

describe("Component: Select", () => {
  test("default snapshot", () => {
    const component = renderer.create(getComponent());
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should open when input is clicked", () => {
    const component = mount(getComponent());

    expect(component.find(Option)).toHaveLength(0);
    component
      .find(`#${SELECT_INPUT_ID}`)
      .hostNodes()
      .simulate("click");

    expect(component.find(Option)).toHaveLength(countries.length);
  });

  test("should open when input is clicked", () => {
    const component = mount(getComponent());

    expect(component.find(Option)).toHaveLength(0);
    component
      .find(`#${SELECT_INPUT_ID}`)
      .hostNodes()
      .simulate("click");

    expect(component.find(Option)).toHaveLength(countries.length);
  });

  test("should allow only digits to be typed / pasted", () => {
    const spy = sinon.spy();
    const component = mount(
      getComponent({
        onChange: spy
      })
    );

    const phoneInput = component.find(`#${PHONE_INPUT_ID}`).hostNodes();
    phoneInput.simulate("change", {
      target: {
        value: "99-997-(876)"
      }
    });
    expect(
      spy.calledWith({
        phone: "99997876",
        countryCode: "+1"
      })
    ).toBeTruthy();
  });

  test("should not trigger unnecessary renders", () => {
    const spy = sinon.spy();
    const component = mount(
      getComponent({
        phone: "99997876",
        onChange: spy
      })
    );

    const phoneInput = component.find(`#${PHONE_INPUT_ID}`).hostNodes();
    phoneInput.simulate("change", {
      target: {
        value: "99997876a"
      }
    });

    expect(spy.notCalled).toBeTruthy();
  });

  test("should be able to empty the input", () => {
    const spy = sinon.spy();
    const component = mount(
      getComponent({
        phone: "99997876",
        onChange: spy
      })
    );
    const phoneInput = component.find(`#${PHONE_INPUT_ID}`).hostNodes();
    phoneInput.simulate("change", {
      target: {
        value: ""
      }
    });

    expect(
      spy.calledWith({
        phone: "",
        countryCode: "+1"
      })
    ).toBeTruthy();
  });

  test("should be able to select country", () => {
    const spy = sinon.spy();
    const component = mount(
      getComponent({
        onChange: spy
      })
    );

    component
      .find(`#${SELECT_INPUT_ID}`)
      .hostNodes()
      .simulate("click");
    component
      .find(Option)
      .at(0)
      .simulate("click");
    expect(
      spy.calledWith({
        phone: "",
        countryCode: countries[0].country_code
      })
    ).toBeTruthy();
  });

  test("controlled input", () => {
    const spy = sinon.spy();
    const component = mount(
      getComponent({
        onChange: spy,
        phone: "998127",
        countryCode: countries[0].country_code
      })
    );

    expect(
      (component
        .find(`#${PHONE_INPUT_ID}`)
        .hostNodes()
        .getDOMNode() as HTMLInputElement).value
    ).toEqual("998127");
    const props = component.find(Select).props();
    expect(props.selected).toEqual(countries[0].country_code);
  });
});
