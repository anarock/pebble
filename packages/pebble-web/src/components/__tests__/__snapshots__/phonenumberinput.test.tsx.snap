// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Select default snapshot 1`] = `
.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  color: #9FA6B0;
  font-size: 14px;
  line-height: 12px;
  position: absolute;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  -webkit-transform: translate(0,24px);
  -ms-transform: translate(0,24px);
  transform: translate(0,24px);
  pointer-events: none;
  z-index: 1;
}

.emotion-0._pebble_input_label_focused {
  font-size: 12px;
  line-height: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  color: #6B7785;
}

.emotion-8 {
  margin-bottom: 20px;
  width: 80px;
  margin-bottom: 0;
}

.emotion-7 {
  position: relative;
}

.emotion-6 {
  cursor: pointer;
  position: relative;
}

.emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ffffff;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
  height: 68px;
  pointer-events: none;
  margin-bottom: 0;
}

.emotion-4._pebble_input_wrapper_textarea {
  height: 110px;
}

.emotion-1 {
  outline: 0;
  border: 0;
  border-bottom: 1px solid #EDEDED;
  padding: 24px 0 12px 0;
  height: 48px;
  border-radius: 0;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
  width: 100%;
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #6B7785;
  color: #101721;
  padding-right: 15px;
}

.emotion-1:disabled {
  background-color: #ffffff;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1:-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::placeholder {
  color: #E0E0E0;
}

.emotion-2 {
  color: #9FA6B0;
  font-size: 14px;
  line-height: 12px;
  position: absolute;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  -webkit-transform: translate(0,24px);
  -ms-transform: translate(0,24px);
  transform: translate(0,24px);
  pointer-events: none;
}

.emotion-2._pebble_input_label_focused {
  font-size: 12px;
  line-height: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  color: #6B7785;
}

.emotion-3 {
  height: 2px;
  background-color: #6161FF;
  margin-top: -2px;
  position: relative;
  width: 0;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  z-index: 9;
}

.emotion-3._pebble_input_highlight_state,
.emotion-3._pebble_input_highlight_focused {
  width: 100%;
}

.emotion-5 {
  position: absolute;
  top: 0;
  bottom: 0;
  margin: auto;
  height: 10px;
  right: 7px;
  color: #9FA6B0;
  font-size: 10px;
}

.emotion-5.__pebble__select__open {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-12 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ffffff;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
  height: 68px;
}

.emotion-12._pebble_input_wrapper_textarea {
  height: 110px;
}

.emotion-9 {
  outline: 0;
  border: 0;
  border-bottom: 1px solid #EDEDED;
  padding: 24px 0 12px 0;
  height: 48px;
  border-radius: 0;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
  width: 100%;
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-9:disabled {
  background-color: #ffffff;
}

.emotion-9::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-9::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-9:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-9:-moz-placeholder {
  color: #E0E0E0;
}

.emotion-9::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-9::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-9:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-9::placeholder {
  color: #E0E0E0;
}

<div
  className="emotion-13"
>
  <label
    className="_pebble_input_label_focused emotion-0"
  >
    Phone No.
  </label>
  <div
    className="emotion-8"
  >
    <div
      className="emotion-7"
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "width": "100%",
          }
        }
      >
        <div
          className="emotion-6"
          onClick={[Function]}
        >
          <div
            className=" emotion-4"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <input
              className="emotion-1"
              id="phone-select-input-test"
              onChange={[Function]}
              readOnly={true}
              value="+1"
            />
            <label
              className="_pebble_input_label_focused emotion-2"
            >
              
            </label>
            <div
              className="_pebble_input_highlight_read_only emotion-3"
              style={
                Object {
                  "backgroundColor": "#6161FF",
                }
              }
            />
            
          </div>
          <i
            className="pi pi-arrow-drop-down emotion-5"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className=" emotion-12"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <input
      className="emotion-9"
      id="phone-test-input-test"
      onChange={[Function]}
      readOnly={false}
      value=""
    />
    <label
      className=" emotion-2"
    >
      
    </label>
    <div
      className=" emotion-3"
      style={
        Object {
          "backgroundColor": "#6161FF",
        }
      }
    />
  </div>
</div>
`;
